docs/src/views/ShorthandProps.tsx

```diff
@@ -1,37 +1,263 @@
+const codeExample = snippet => <CodeSnippet value={joinToString(snippet)} />
+
 class ShorthandProps extends React.Component {
   render() {
     return (
       <DocPage title="Shorthand Props">
         <p>
-          Most components support two APIs: <code>shorthand</code> and <code>children</code>. With
-          the <code>shorthand</code> definition we are adding some props to the component in order
-          to define it’s content and behavior.
+          It is quite common for Stardust component to have "slots" which accept shorthand values.
+          For example, {code('Button')} component has an {code('icon')} slot which value defines the
+          icon that will be rendered.
+        </p>
+
+        {codeExample([`<Button icon='chess rook' />`])}
+
+        <p>
+          So, what this string value (i.e. {code("'chess rook'")}) means for {code('Button')}{' '}
+          component and, essentially, how it transforms into rendered {code('<Icon />')} element?
+        </p>
+
+        <p>
+          There are several forms of shorthand values that can be provided, but all of them share
+          one common thing - each is eventually evaluated to React Element. Thus, you can think of
+          shorthand values as a recipe to customize rendered React Element at corresponding slot.
+        </p>
+
+        {/* SHORTHAND AS AN OBJECT */}
+        {header('Shorthand value as Object')}
+        <p>
+          Each component's slot has associated default element type. For example, by default there
+          is {code('<Icon />')} element rendered for {code('Button')}'s {code('icon')} slot. It is
+          possible to customize props of this default element by providing props object as shorthand
+          value:
+        </p>
+
+        {codeExample([
+          `// 'name' and 'size' will be used as <Icon /> element's props`,
+          `<Button icon={{ name: 'chess rook', size: 'small' }} />`,
+        ])}
+
+        {/* SHORTHAND AS A PRIMITIVE VALUE */}
+        {header('Shorthand value as Primitive')}
+        <p>
+          There is even shorter way to define default element's props - by using a primitive value.
+          In that case provided shorthand value will be taken as a value for 'default prop' of this
+          default element.
+        </p>
+
+        <p>
+          This could be much easier seen with an example. Here, again, we have a {code('Button')}{' '}
+          element with its icon being defined with shorthand - where provided {code('string')} value
+          will be used as icon's {code('name')}:
+        </p>
+
+        {codeExample([
+          `<>`,
+          `  <Button icon='chess rook' />`,
+          ``,
+          `  {/* has identical effect to the previous one */}`,
+          `  <Button icon={{ name: 'chess rook' }} />`,
+          `</>`,
+        ])}
+        <p>
+          This works because {code('name')} is the default prop of slot's {code('<Icon />')}{' '}
+          element.
```

Comment:
This works because {code('name')} is the default prop of the {code('<Icon />')} component.

Reply:
a bit reluctant to use this terminology, as `default props` for Component may lead to false associations. Essentially, default element's prop is even more correct, given how this default prop's value is handled